xbmc/video/tags/ISetInfoTagLoader.h

```diff
@@ -0,0 +1,61 @@
+/*
+ *  Copyright (C) 2017-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "InfoScanner.h"
+#include "addons/Scraper.h"
+
+#include <string>
+#include <utility>
+
+class CFileItem;
+class CSetInfoTag;
+class EmbeddedArt;
+
+namespace VIDEO
+{
+
+//! \brief Base class for set tag loaders.
+class ISetInfoTagLoader
+{
+public:
+  //! \brief Constructor
+  //! \param item The item to load info for
+  //! \param info Scraper info
+  //! \param llokInFolder True to look in folder holding file
+  ISetInfoTagLoader(const CFileItem& item, ADDON::ScraperPtr info, bool lookInFolder = false)
+    : m_item(item), m_info(std::move(info))
```

Comment:
Sometimes you are using the copy and move idiom, sometimes param as reference. Is this by intention or just copy/paste from exiting code?

Reply:
Sorry, just copy/paste existing code. I wanted to keep it as consistent as possible.